test/emf/emf_container_test.go

```diff
@@ -29,37 +31,51 @@ func (t *EMFECSTestRunner) Validate() status.TestGroupResult {
 	}
 }
 
-func (t *EMFECSTestRunner) GetTestName() string {
-	return "EMFOnECS"
+func (t *EMFTestRunner) GetTestName() string {
+	return "EMF Container Tests"
 
 }
 
-func (t *EMFECSTestRunner) GetAgentConfigFileName() string {
+func (t *EMFTestRunner) GetAgentConfigFileName() string {
 	return "./resources/config.json"
 }
 
-func (t *EMFECSTestRunner) GetAgentRunDuration() time.Duration {
+func (t *EMFTestRunner) GetAgentRunDuration() time.Duration {
 	return 3 * time.Minute
 }
 
-func (t *EMFECSTestRunner) GetMeasuredMetrics() []string {
+func (t *EMFTestRunner) GetMeasuredMetrics() []string {
 	return []string{"EMFCounter"}
 }
 
-func (t *EMFECSTestRunner) validateEMFOnECSMetrics(metricName string) status.TestResult {
+func (t *EMFTestRunner) validateEMFMetrics(metricName string) status.TestResult {
+	namespace := ""
+	var dims []types.Dimension
+	var failed []dimension.Instruction
+	if t.testName == "EMF_ECS" {
+		namespace = "EMFECSNameSpace"
+		dims, failed = t.DimensionFactory.GetDimensions([]dimension.Instruction{
+			{
+				Key:   "InstanceName",
+				Value: dimension.UnknownDimensionValue(),
+			},
+		})
+	}
+	if t.testName == "EMF_EKS" {
+		namespace = "EMFEKSNameSpace"
+		dims, failed = t.DimensionFactory.GetDimensions([]dimension.Instruction{})
+	}
```

Comment:
I don't like using the test name to switch on what we configure for validation. I'd prefer if we could make this a runner implementation, but I have a feeling that our runner interface is already divergent and getting bloated.